plugins/bufferlines/barbar.nix

```diff
@@ -69,249 +24,566 @@ let
+    "focusOnClose"
+    [
+      "hide"
+      "alternate"
+    ]
+    [
+      "hide"
+      "current"
+    ]
+    [
+      "hide"
+      "extensions"
+    ]
+    [
+      "hide"
+      "inactive"
+    ]
+    [
+      "hide"
+      "visible"
+    ]
+    "highlightAlternate"
+    "highlightInactiveFileIcons"
+    "highlightVisible"
+    [
+      "icons"
+      "bufferIndex"
+    ]
+    [
+      "icons"
+      "bufferNumber"
+    ]
+    [
+      "icons"
+      "button"
+    ]
+    [
+      "icons"
+      "filetype"
+      "customColors"
+    ]
+    [
+      "icons"
+      "separator"
+      "left"
+    ]
+    [
+      "icons"
+      "separator"
+      "right"
+    ]
+    "insertAtStart"
+    "insertAtEnd"
+    "maximumPadding"
+    "minimumPadding"
+    "maximumLength"
+    "semanticLetters"
+    "letters"
+    "sidebarFiletypes"
+    "noNameTitle"
+    "tabpages"
+  ];
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "barbar"
+      ];
+      settingsPath = basePluginPath ++ [ "settings" ];
+    in
+    [
+      (mkRenamedOptionModule (basePluginPath ++ [ "excludeFileTypes" ]) (
+        settingsPath ++ [ "exclude_ft" ]
+      ))
+      (mkRenamedOptionModule (basePluginPath ++ [ "excludeFileNames" ]) (
+        settingsPath ++ [ "exclude_name" ]
+      ))
+      (mkRemovedOptionModule (
+        basePluginPath
+        ++ [
+          "icons"
+          "diagnostics"
+        ]
+      ) "Use `settings.icons.diagnostics` instead, but pay attention as the keys have changed.")
+      (mkRenamedOptionModule
+        (
+          basePluginPath
+          ++ [
+            "icons"
+            "filetype"
+            "enable"
+          ]
+        )
+        (
+          settingsPath
+          ++ [
+            "icons"
+            "filetype"
+            "enabled"
+          ]
+        )
+      )
+    ]
+    ++ (map
+      (
+        name:
+        mkRemovedOptionModule (
+          basePluginPath
+          ++ [
+            "icons"
+            name
+          ]
+        ) "Use `settings.icons.${name}` instead, but you should now use the real `snake_case` key names."
+      )
+      [
+        "alternate"
+        "current"
+        "inactive"
+        "modified"
+        "pinned"
+        "visible"
+      ]
+    );
+
+  extraOptions = {
+    keymaps =
+      {
+        silent = mkEnableOption "silent keymaps for barbar";
+      }
+      // (mapAttrs (
+        optionName: funcName: helpers.mkNullOrOption types.str "Keymap for function Buffer${funcName}"
+      ) keymapsActions);
+  };
 
-    package = helpers.mkPluginPackageOption "barbar" pkgs.vimPlugins.barbar-nvim;
+  extraConfig = cfg: {
+    keymaps = flatten (
+      mapAttrsToList (
+        optionName: funcName:
+        let
+          key = cfg.keymaps.${optionName};
+        in
+        optional (key != null) {
+          mode = "n";
+          inherit key;
+          action = "<Cmd>Buffer${funcName}<CR>";
+          options.silent = cfg.keymaps.silent;
```

Comment:
`<cmd>` is always silent anyway

Reply:
Good to know. I have removed the option.